From fb5eaf13a742fb0debb47f65400ae3073122b7d0 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Mon, 30 Nov 2015 11:39:13 +0000
Subject: [PATCH 180/195] staging: etnaviv: remove locking in
 etnaviv_ioctl_gem_info()

Re-using the commit message from Daniel Vetter for Armada DRM:

  Since David Herrmann's mmap vma manager rework we don't need to grab
  dev->struct_mutex any more to prevent races when looking up the mmap
  offset.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_drv.c | 8 +-------
 drivers/staging/etnaviv/etnaviv_gem.c | 1 -
 2 files changed, 1 insertion(+), 8 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_drv.c b/drivers/staging/etnaviv/etnaviv_drv.c
index 6b6e5c1..8b91eae 100644
--- a/drivers/staging/etnaviv/etnaviv_drv.c
+++ b/drivers/staging/etnaviv/etnaviv_drv.c
@@ -370,13 +370,7 @@ static int etnaviv_ioctl_gem_info(struct drm_device *dev, void *data,
 	if (!obj)
 		return -ENOENT;
 
-	ret = mutex_lock_interruptible(&dev->struct_mutex);
-	if (ret == 0) {
-		ret = etnaviv_gem_mmap_offset(obj, &args->offset);
-
-		mutex_unlock(&dev->struct_mutex);
-	}
-
+	ret = etnaviv_gem_mmap_offset(obj, &args->offset);
 	drm_gem_object_unreference_unlocked(obj);
 
 	return ret;
diff --git a/drivers/staging/etnaviv/etnaviv_gem.c b/drivers/staging/etnaviv/etnaviv_gem.c
index 46403b6..8655e60 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.c
+++ b/drivers/staging/etnaviv/etnaviv_gem.c
@@ -229,7 +229,6 @@ out:
 	}
 }
 
-/* get mmap offset - must be called under struct_mutex */
 int etnaviv_gem_mmap_offset(struct drm_gem_object *obj, u64 *offset)
 {
 	int ret;
-- 
2.6.2

